src/views/docs-view/utils/set-document-path.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { isReleaseNotesPage } from 'lib/docs/is-release-notes-page'
+import path from 'node:path'
+import { getValidVersions } from './get-valid-versions'
+import { VersionSelectItem } from '../loaders/remote-content'
+
+export async function fetchValidVersions(
```

Comment:
An observation and opinion about utils. This function is used once, and in general I become wary of structuring code around a utils pattern as it can become kind of a junk drawer whereas a function used once could also be co-located with the code that calls it. Nothing to change here, but could be a topic for 'how we work' for some later date.

Reply:
That's a great point. My thinking here was that it would be easier to test if this functionality was moved out of server.ts, but I suppose I could define the function at the top of the file and export it so it can still be tested. WDYT?

Reply:
I would move the function to the top of server.ts and export it. One less file, one less import, one less context jump for the future developer.